dist/tools/usb-serial/ttys.py

```diff
@@ -194,16 +173,45 @@ def print_results(args, ttys):
         print(tty[args.format])
 
 
+def generate_filters(args):
+    """
+    Generate filters for use in the filters_match function from the command
+    line arguments
+    """
+    result = list()
+    if args.serial is not None:
+        result.append(("serial", re.compile(r"^" + re.escape(args.serial)
+                       + r"$")))
```

Comment:
> If the intention is that serial numbers stay a precise match, you may want to pass the pattern through re.escape.

I thought this did this. Anything I missed?

> But I don't expect any practical serial numbers to be affected by that

Note that a serial `0001` as regex would also match also e.g. `acd0001f`. I think it is better to not count on that.

Reply:
> Anything I missed?

No, my testing was erroneous; I had switched from serials on to vendors by the time I did the top-anchoring tests, and misremembered an outcome.

> Note that a serial `0001`

Yes -- but most serials contain enough entropy for this not to happen.